yt_dlp/extractor/lastfm.py

```diff
@@ -0,0 +1,158 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    int_or_none
+)
+
+
+class LastFMPlaylistBaseIE(InfoExtractor):
+    def _extract_current_page(self, url):
+        return int_or_none(self._search_regex(
+            r'\bpage=(\d+)', url, 'page', default=None))
+
+    def _extract_last_page(self, webpage):
+        last_page = None
+        pagination_list = self._search_regex(
+            r'(?s)class="[^"]*pagination-list[^"]*"[^>]*>(.+?)</ul>', webpage, 'pagination_list', fatal=False,
+            default=None)
+
+        if pagination_list:
+            page_numbers = re.findall(r'(\d+)', pagination_list)
+            last_page = max([int_or_none(page_number) for page_number in page_numbers])
```

Comment:
this is fatal
```sh
>>> max([1, None, 2, None, 23])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '>' not supported between instances of 'NoneType' and 'int
```

Reply:
`\d+` will only match integers here, so this case is invalid

Reply:
```py
last_page = max(re.findall(r'(\d+)', pagination_list))
```

is sufficient here. int_or_none isnt doing anything

Reply:
i beg to differ. `max(re.findall(r'(\d+)', '1 2 3 4 10'))` will return `4` which is clearly not the last page

Reply:
Correct. At least it must be something like `max(map(int_or_none, re.findall(r'(\d+)', '1 2 3 4 10')))`

Reply:
right, my bad